marklogic-data-hub-central/ui/src/components/explore/entity-type-display-settings-modal/entity-type-display-settings-modal.tsx

```diff
@@ -54,8 +65,35 @@ const EntityTypeDisplaySettingsModal: React.FC<Props> = ({isVisible, hubCentralC
     return options;
   };
 
-  const onEntityLabelChange = (row, e) => {
-    setEntityLabels({...entityLabels, [row.entityType]: e});
+  const onColumnValueChange = (row, e, column: TableColumns) => {
+    setExploreSettingsData(exploreSettingsData => {
+      let settingsData = exploreSettingsData.map(entityData => Object.assign({}, entityData));
```

Comment:
I would avoid iterating over the entire array every time we are changing Icon, color label or propertiesOnHover. 
Perhaps keeping them separately in either one object with structure similar to "hubCentralConfig.modeling.entities" or in separate objects for each column (like we are doing in few other areas in HC might help. That would really help when the database scales up to include more that 100-1500 entities.

Reply:
**For Example**: If you create a separate state object variable (may be "settingsData") to store all metadata, then your logic here can be reduced to something like below.

`let tempSettingData = settingsData;
    let exploreSettingRow = tempSettingData[row.entityType];
    switch (column) {
          case TableColumns.Color:
          exploreSettingRow.color = e.color.hex;
          break;
        case TableColumns.Icon:
          exploreSettingRow.icon = e;
          break;
        case TableColumns.EntityLabel:
          exploreSettingRow.label = e.value;
          break;
        }
        setSettingsData(tempSettingsData);`

And then we can use this "settingsData" object to be refresh table columns and also avoid iterating over the Array in handleSave by doing something like this 
"updatedPayload.modeling.entities = exploreSettingsData"
This is just a high level layout of the approach and of course there would be some other adjustments in the code.